### PR TITLE
PhysicsManager CHECK cleanup

### PR DESCRIPTION
**Pull request by aclegg3**
## Motivation and Context

This PR replaces if statements in PhysicsManager and BulletPhysicsManager based on the validity of object IDs with CHECK().

As @msbaines mentioned in a review for PR #237, PhysicsManager functions taking an object ID as input should not fail silently when an invalid object ID is provided. This may allow hidden bugs to crop up. For example: setting a force or torque on an object which does not exist and then receiving no indication that the set failed resulting in extended debugging. Instead, such an invalid set should be fatal and produce a clear error message. 

## How Has This Been Tested

Local testing.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/esp/physics/PhysicsManager.cpp [hunk 1]
```diff
@@ -84,10 +84,7 @@ int PhysicsManager::addObject(const std::string& configFile,
 }
 
 int PhysicsManager::removeObject(const int physObjectID) {
-  if (existingObjects_.count(physObjectID) == 0) {
-    LOG(ERROR) << "Failed to remove object: no object with ID " << physObjectID;
-    return ID_UNDEFINED;
-  }
+  assertIDValidity(physObjectID);
```
Comment (msbaines) [inline src/esp/physics/PhysicsManager.cpp]: Should the object destructor call removeObject()? That way you never forget to call it.
delete should be avoid and RTTI used instead. Could we use unique_ptr instead here? The last three lines would then boil down to:

existingObjects_.erase(physObjectID);

Comment (aclegg3) [reply src/esp/physics/PhysicsManager.cpp]: This comes back to the SceneNode problem. SceneNode objects are owned completely by the SceneGraph. I have not gotten around to converting RigidObject from a SceneNode to an AbstractFeature3D as suggested in reviews for PR #91 . We use raw pointers to denote that PhysicsManager does not own the RigidObject. 

That said, the removal process could probably be improved. I'll lump that in with the AbstractFeature3D re-design.

Comment (msbaines) [reply src/esp/physics/PhysicsManager.cpp]: If we don't own the object, why are we calling delete on it below on line 89?

Comment (aclegg3) [reply src/esp/physics/PhysicsManager.cpp]: We have to remove the SceneNode from the SceneGraph when the object no longer exists. I agree the current situation is not entirely self-consistent. The simulator was not initially designed for this sort of dynamic restructuring.

**Pull request merged**